### PR TITLE
Adds TransformStreamDefaultController

### PR DESCRIPTION
**Pull request by rachelandrew**
Documenting the `TransformStreamDefaultController` interface.

I also updated the main Streams API overview to link to this interface, and TransformStream. In working on this I saw that TransformStream is missing a bunch of pages: https://developer.mozilla.org/en-US/docs/Web/API/TransformStream

Reviewer: @jpmedley

### CODE REVIEW

File: files/en-us/web/api/transformstreamdefaultcontroller/enqueue/index.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+<h3 id="Parameters">Parameters</h3>
+
+<dl>
+  <dt><code>chunk</code></dt>
+  <dd>The chunk being queued. A chunk is a single piece of data. It can be any type of data, and a stream can contain chunks of different types.</dd>
+</dl>
+
+<h2 id="Examples">Examples</h2>
+
+<p>In this example an endoded string is passed to the queue using the <code>enqueue()</code> method.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/transformstreamdefaultcontroller/enqueue/index.html:30]:
```suggestion
<p>In this example an encoded string is passed to the queue using the <code>enqueue()</code> method.</p>
```

File: files/en-us/web/api/transformstreamdefaultcontroller/enqueue/index.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+<pre class="brush: js">controller.enqueue(this.textencoder.encode(String(chunk)))</pre>
```
Comment (jpmedley) [inline files/en-us/web/api/transformstreamdefaultcontroller/enqueue/index.html:32]: Is `controller` an instance of `ReadableStream`. Please make it clear what this is.

File: files/en-us/web/api/transformstreamdefaultcontroller/error/index.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  - error
+  - TransformStreamDefaultController
+---
+<div>{{DefaultAPISidebar("Streams API")}}</div>
+
+<p class="summary">The <strong><code>error()</code></strong> method of the {{domxref("TransformStreamDefaultController")}} interface errors both sides of the stream. Any further interactions with it will fail with the given error message, and any chunks in the queue will be discarded.</p>
+
+<h2 id="Syntax">Syntax</h2>
+
+<pre class="syntaxbox">TransformStreamDefaultController.error(e);</pre>
```
Comment (jpmedley) [inline files/en-us/web/api/transformstreamdefaultcontroller/error/index.html:17]:
```suggestion
<pre class="syntaxbox">TransformStreamDefaultController.error(reason);</pre>
```

File: files/en-us/web/api/transformstreamdefaultcontroller/error/index.html [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+<h3 id="Parameters">Parameters</h3>
+
+<dl>
+  <dt><code>e</code></dt>
```
Comment (jpmedley) [inline files/en-us/web/api/transformstreamdefaultcontroller/error/index.html:22]:
```suggestion
  <dt><code>reason</code></dt>
```

File: files/en-us/web/api/transformstreamdefaultcontroller/error/index.html (hunk 3 above)
Comment (jpmedley) [reply files/en-us/web/api/transformstreamdefaultcontroller/error/index.html:17]: The IDL term seems useful here. 'Reason' is more descriptive than 'e'.

File: files/en-us/web/api/transformstreamdefaultcontroller/index.html [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  - API
+  - Interface
+  - Reference
+  - TransformStreamDefaultController
+---
+<div>{{DefaultAPISidebar("Streams API")}}</div>
+
+<p class="summary">The <strong><code>TransformStreamDefaultController</code></strong> interface of the {{domxref('Streams API','','',' ')}} provides methods to manipulate the associated {{domxref("ReadableStream")}} and {{domxref("WritableStream")}}.</p>
+
+<p>When constructing a {{domxref("TransformStream")}}, the <code>TransformStreamDefaultController</code> is created. It therefore has no constructor.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: Please describe how to get an instance of this.

**Review by jpmedley — CHANGES_REQUESTED** [review]
Typo wise, this is really clean. I have a few conceptual questions that I think users will have an so need to be answered.

File: files/en-us/web/api/transformstreamdefaultcontroller/desiredsize/index.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+tags:
+  - API
+  - Property
+  - Reference
+  - desiredSize
+  - TransformStreamDefaultController
+---
+<div>{{DefaultAPISidebar("Streams API")}}</div>
+
+<p class="summary">The <strong><code>desiredSize</code></strong> read-only property of the {{domxref("TransformStreamDefaultController")}} interface returns the desired size to fill the queue of the associated {{domxref("ReadableStream")}}.</p>
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply files/en-us/web/api/transformstreamdefaultcontroller/desiredsize/index.html:13]: No it's letting you know the state of the readable stream so you can apply backpressure or do some other management https://developer.mozilla.org/en-US/docs/Web/API/Streams_API/Concepts#backpressure in my next commit I've added a link to that.

File: files/en-us/web/api/transformstreamdefaultcontroller/index.html (hunk 5 above)
Comment (rachelandrew) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: I'm having trouble doing so without using the terminology "Internal slots" of which you are not fond https://streams.spec.whatwg.org/#ts-internal-slots it doesn't help that TransformStream itself isn't documented as I actually think this is better detailed there, in the same way as ReadableStream https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream/ReadableStream

It's created by constructing a TransformStream, which I already have. The details of that really belong to TransformStream.

File: files/en-us/web/api/transformstreamdefaultcontroller/enqueue/index.html (hunk 2 above)
Comment (rachelandrew) [reply files/en-us/web/api/transformstreamdefaultcontroller/enqueue/index.html:32]: We're dealing with `TransformStreamDefaultController` so it's an instance of that, as part of the transformer. I found a succinct example that maybe is better. Adding in my next commit.

File: files/en-us/web/api/transformstreamdefaultcontroller/index.html (hunk 5 above)
Comment (jpmedley) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: You should reach out to the spec author. If there's not a constructor, there should be a property on another object that returns it. I can't find one anywhere.

Comment (rachelandrew) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: sorry I totally missed this, I think it was during the period I also had all the Learn CSS PRs to wrangle and it all got a bit noisy.

I've just had another look at this to refresh my memory, and I'm not really sure what I'd be asking a spec author that isn't already in the spec (so I think I don't understand what the problem you have here is, I must be missing something).

In https://streams.spec.whatwg.org/#ts-default-controller-abstract-ops it seems clear that the TransformStreamDefaultController is created during the setup process of a TransformStream, I think it needs documenting in the missing pages of TransformStream. But if you can explain again what I should ask a spec author, that isn't what is detailed there I'm happy to, I just don't understand what I'd be asking them right now :)

Comment (jpmedley) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: I'm asking about `TransformStreamDefaultController`. You state that it's created when constructing a `TransformStream`. I can't find a method or property anywhere that returns it. It doesn't have a constructor either. How do I get an instance of it?

Comment (rachelandrew) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: I've been chatting to people in the whatwg irc about this, and the only way to gain access is via the Transformer callbacks https://streams.spec.whatwg.org/#transformer-api

So, I still think this is better detailed in the TransformStream constructor docs, as that is how a developer would be accessing it. Those docs don't exist as there is just an interface page, but I could write it.

Comment (jpmedley) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html:14]: Please write it. Then add a quick line about getting it from the TransformStream with a link to the details.

Comment (rachelandrew) [reply files/en-us/web/api/transformstreamdefaultcontroller/index.html]: I've created a separate PR for the missing TransformStream pages, and added a line to the main page for this one.

**Review by jpmedley — APPROVED** [review]

**Comment by jpmedley** [comment]
I'm not sure why this is failing, and I'm not sure which of our colleagues to ask for help.

**Pull request merged**